game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProTerritoryValueUtils.java

```diff
@@ -177,19 +176,31 @@ public static double findTerritoryAttackValue(
     return territoryValueMap;
   }
 
-  private static int findMaxLandMassSize(final GamePlayer player) {
-    int maxLandMassSize = 1;
+  protected static int findMaxLandMassSize(final GamePlayer player) {
     final GameData data = player.getData();
+    final Predicate<Territory> cond = ProMatches.territoryCanPotentiallyMoveLandUnits(player, data);
+    class LandMassFinder extends BreadthFirstSearch {
+      int landMassSize = 0;
+
+      LandMassFinder(final Territory startTerritory) {
+        super(startTerritory, cond);
+      }
+
+      @Override
+      public void visit(final Territory territory) {
+        landMassSize++;
+      }
+    }
```

Comment:
Ok, I didn't know you could nest classes inside methods 😮

Reply:
Yeah, not really sure how I feel about classes nested in a method either...

Reply:
It was to support tracking of landMassSize while keeping the logic self contained to the method. The other option is an anon class and have:

final int maxLandMassSize[] = new int[1];

in the method scope so that the inner class can write to maxLandMassSize[0] (since you can't write to an int. This seemed cleaner than that.

Reply:
Would `AtomicReference` perhaps help? I think that can be templated to an integer, eg: `AtomicReference<Integer>`

Reply:
It would work, but the atomicity is not needed and may be confusing to readers.

Reply:
`AtomicReference` AFAIK is preferred way to avoid the array of size 1:
`final int maxLandMassSize[] = new int[1];`

There might be other solutions to be had too..

Reply:
Do you have a reference for why it's preferred?

It seems an antipattern to me. We're using something that was not meant for the purpose and has a semantic meaning (i.e. for use with multithreading) for something else. This both confuses the semantic meaning (a reader may think the code is multithreaded when it's not) and also can have performance implications.

Imho, the named inner class is the cleanest way to do without a dedicated MutableInt class (like exists in apache commons).

I searched a few sources to see if there's a reason to prefer AtomicInteger, but haven't found anything convincing:
https://www.techiedelight.com/pass-integer-reference-java/
https://stackoverflow.com/questions/319765/accessing-inner-anonymous-class-members

Reply:
Either way, `int[]` or `AtomicReference<Integer>` or `AtomicInteger`, for the sake of using a final value that is mutable, it's not the intended usage of any of those.

I know I've seen `AtomicReference` used for that before, don't have any examples stating it's a best practice (AFAIK none of it is necessarily a best practice though as we are hacking around a variable that needs to be final).

Reply:
Well, the other approach was to have an inner class with a member int for, which is what my original change did. I would argue that is less hacky than the others, but then people didn't like a non-anomymous inner class inside a method.  ¯\_(ツ)_/¯

I went with int[] now.

Reply:
Apologies as you're suffering from too many reviewers :grin: 
We've different tastes.
I would probably look for perhaps a 3rd option.
I respect avoiding code churn, `int[]` or `AtomicReference` are not that much different, not a big deal for me.